### PR TITLE
feat: show editor tools in user settings

### PR DESCRIPTION
**Pull request by GeekaN2**
Show list of user editor tools. 
Default ones are included in user editor tools.

Allows to add tools to user by id in his settings.

### CODE REVIEW

File: src/infrastructure/storage/user.ts [hunk 1]
```diff
@@ -9,6 +10,8 @@ export type UserStoreData = {
    * User data
    */
   user: User | null;
+
```
Comment (neSpecc) [inline src/infrastructure/storage/user.ts:13]: docs missed

File: src/infrastructure/storage/user.ts [hunk 2]
```diff
@@ -30,4 +33,20 @@ export class UserStore extends SubscribableStore<UserStoreData> {
...
+  /**
+   * array of tools
+   */
+  public getUserEditorTools(): EditorTool[] {
+    return this.data.editorTools;
+  }
+
+  /**
+   * Setter
+   *
```
Comment (neSpecc) [inline src/infrastructure/storage/user.ts:46]: nasty docs

File: src/application/services/useAppState.ts [hunk 3]
```diff
@@ -11,6 +12,8 @@ interface UseAppStateComposable {
    * Current authenticated user
    */
   user: Ref<User | null>;
+
+  userEditorTools: Ref<EditorTool[]>
```
Comment (neSpecc) [inline src/application/services/useAppState.ts:16]: document please

File: src/domain/user.repository.interface.ts [hunk 4]
```diff
@@ -13,4 +14,8 @@ export default interface UserRepositoryInterface {
    * Return stored user data
    */
   getUser: () => User | null;
+
+  loadUserEditorTools: () => Promise<void>;
+
+  getUserEditorTools: () => EditorTool[];
```
Comment (neSpecc) [inline src/domain/user.repository.interface.ts:20]: docs pls

File: src/domain/entities/EditorTool.ts [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  /**
+   * User-friendly plugin title
+   */
+  name: string;
+
+  /**
+   * Name of the tool class. Since it's imported globally,
+   * we need the class name to properly connect the tool to the editor
+   */
+  class: string;
```
Comment (neSpecc) [inline src/domain/entities/EditorTool.ts:24]: bad naming

**Review by neSpecc — CHANGES_REQUESTED** [review]
Comments ignored

**Review by neSpecc — APPROVED** [review]

**Review by slaveeks — APPROVED** [review]

**Pull request merged**